lib/Controller/Report.php

```diff
@@ -0,0 +1,520 @@
+
+        // Return
+        $this->getState()->hydrate([
+            'httpStatus' => 204,
+            'message' => sprintf(__('Deleted %s'), $reportSchedule->name)
+        ]);
+    }
+
+    /**
+     * Displays the Report Schedule Page
+     */
+    public function displayReportSchedulePage()
+    {
+        // Call to render the template
+        $this->getState()->template = 'report-schedule-page';
+        $this->getState()->setData([
+            'reportSchedules' => $this->reportScheduleFactory->query(),
+        ]);
+    }
+
+    /**
+     * Displays an Add form
+     */
+    public function addReportScheduleForm()
+    {
+
+        $type = $this->getSanitizer()->getParam('type', 'layout');
+        $reportName = $this->getSanitizer()->getParam('reportName', null);
+
+        $data = ['filters' => []];
+
+        $data['filters'][] = ['name'=> 'Daily', 'filter'=> 'daily'];
+        $data['filters'][] = ['name'=> 'Weekly', 'filter'=> 'weekly'];
+        $data['filters'][] = ['name'=> 'Monthly', 'filter'=> 'monthly'];
+        $data['filters'][] = ['name'=> 'Yearly', 'filter'=> 'yearly'];
+        $data['type'] = $type;
+        $data['reportName'] = $reportName;
+
+        if ($type == 'layout') {
+            $data['selectedId'] = $this->getSanitizer()->getParam('layoutId', null);
+            $data['selectedName'] = $this->layoutFactory->getById($data['selectedId'])->layout;
+        } else {
+            $data['selectedId'] = $this->getSanitizer()->getParam('mediaId', null);
+            $data['selectedName'] = $this->mediaFactory->getById($data['selectedId'])->name;
+        }
+
+        $this->getState()->template = 'report-schedule-form-add';
+        $this->getState()->setData($data);
+
+    }
+
+    /**
+     * Report Schedule Edit Form
+     */
+    public function editReportScheduleForm($reportScheduleId)
+    {
+        $reportSchedule = $this->reportScheduleFactory->getById($reportScheduleId);
+
+        $this->getState()->template = 'reportschedule-form-edit';
+        $this->getState()->setData([
+            'reportSchedule' => $reportSchedule
+        ]);
+    }
+
+    /**
+     * Report Schedule Delete Form
+     */
+    public function deleteReportScheduleForm($reportScheduleId)
+    {
+        $reportSchedule = $this->reportScheduleFactory->getById($reportScheduleId);
+
+        if (!$this->getUser()->checkDeleteable($reportSchedule))
+            throw new AccessDeniedException(__('You do not have permissions to delete this report schedule'));
+
+//         var_dump($reportSchedule->name);
+//         die();
+
+        $data = [
+            'reportSchedule' => $reportSchedule
+        ];
+
+        $this->getState()->template = 'reportschedule-form-delete';
+        $this->getState()->setData($data);
+
+    }
+
+    //</editor-fold>
+
+    //<editor-fold desc="Saved report">
+
+    /**
+     * Saved report Grid
+     */
+    public function savedReportGrid()
+    {
+        $reportName = 'summaryReport';
```

Comment:
What is this used for?

Reply:
We need the reportName to be passed as parameter to preview a saved report. I have changed the hard coded reportName.